docs/sphinx/dev_guide/release_process.rst

```diff
@@ -0,0 +1,137 @@
+     bugfixes, etc. in the release. These should be a high-level summary of the 
+     contents of the ``RELEASE_NOTES.md`` file in the RAJA repo, which may 
+     contain more detailed information. Also, add a note to the 
+     release description to remind users to download the gzipped tarfile for 
+     the release, if they will not checkout the tag for the release for some 
+     reason, rather than one of the assets GitHub creates for the release. 
+     The GitHub-created assets do not contain the RAJA submodules and will
+     cause issues for users as a result.
+
+     .. important:: For consistency, please follow a similar description 
+                    pattern for all RAJA releases.
+
+  #. Check out the main branch locally and make sure it is up-to-date.     
+     Then, generate the release tarfile for the release by running the script 
+     ``./scripts/make_release_tarball.sh`` from the top-level RAJA directory. 
+     If this is successful, a gzipped tarfile whose name includes the release 
+     tag **with no extraneous SHA-1 hash information** will be in the top-level
+     RAJA directory.
+  #. Edit the release in GitHub and upload the tarfile to the release.
+  #. Make a PR to merge the main branch into the develop branch. After it 
+     passes all CI checks and is approved, merge the PR. This will ensure that
+     all changes done to finalize the release will not be lost in future
+     changes to the develop branch.
+
+.. _rcbranch-label:
+
+===========================
+Release Candidate Branch
+===========================
+
+A *release candidate* branch is a temporary branch used to finalize a release.
+When the features, documentation, bug fixes, etc.  to include in a release are 
+complete and merged into the develop branch, a release candidate branch is made
+off of the develop branch. Typically, a release candidate branch is named 
+**rc-<release #>.** Please see :ref:`release-label` for a description of how 
+a release candidate branch is used in the release process. 
+
+.. important:: Creating a release candidate branch starts the next release 
+               cycle whereby new work being performed on feature branches can 
+               be merged into the develop branch.
+
+Finalizing a release on a release candidate branch involves the following steps:
+
+  #. If not already done, create a section for the release in the RAJA
+     ``RELEASE_NOTES.md`` file. Describe all API changes, notable new features,
+     bug fixes, improvements, build changes, etc. included in the release in 
+     appropriately labeled sections of the file. Please follow the pattern
+     established in the file for previous releases. All changes that users 
+     should be aware of should be documented in the release notes. Hopefully,
+     the release notes file has been updated along with the corresponding
+     changes as they are merged into the develop branch. At any rate, it is
```

Comment:
Umpire makes a "changelog" update required to merge the branch - maybe y'all could do that too

Reply:
I've tried. I've caved.

Reply:
Is there a way to extract all the PR descriptions easily into one spot? Because if people are better about making good PR descriptions when initially creating the PR, maybe that could also be a way of keeping track of cool changes.

Reply:
Possibly. Our PR descriptions tend to contain much more detail than I think is appropriate for release notes. Not sure how to automatically edit down.

Reply:
Do the PRs that correspond to each release get marked some how? Like with Umpire, we can see that certain PRs (really JIRA tickets) correspond with certain releases. That could at least narrow down a list of PRs can someone could skim through really quickly before confirming that the release notes are updated.. just an idea :p

Reply:
Yup. Could use GitHub milestone feature. But, that's more planning and thinking ahead than I can manage.

Reply:
Lol! Well in that case..